forge/src/main/java/org/openjdk/skara/forge/gitlab/GitLabForgeFactory.java

```diff
@@ -43,6 +47,33 @@ public String name() {
         return Set.of("gitlab.com");
     }
 
+    private void configureSshKey(String userName, String hostName, String sshKey) {
+        var cfgPath = Path.of(System.getProperty("user.home"), ".ssh");
+        var cfgFile = cfgPath.resolve("config");
+        var existing = "";
+        try {
+            existing = Files.readString(cfgFile, StandardCharsets.UTF_8);
+        } catch (IOException ignored) {
+        }
+
+        var userHost = userName + "." + hostName;
+        var existingBlock = Pattern.compile("^Match host=" + Pattern.quote(userHost) + "(?:\\R[ \\t]+.*)+", Pattern.MULTILINE);
+        var existingMatcher = existingBlock.matcher(existing);
+        var filtered = existingMatcher.replaceAll("");
+        var result = "Match host=" + userHost + "\n" +
+                "  Hostname " + hostName + "\n" +
+                "  PreferredAuthentications publickey\n" +
+                "  StrictHostKeyChecking no\n" +
+                "  IdentityFile " + sshKey + "\n" +
+                "\n";
+
+        try {
+            Files.writeString(cfgFile, result + filtered.strip() + "\n", StandardCharsets.UTF_8);
```

Comment:
Can this result in us writing to a users .ssh/config when running Skara CLI tools?

Reply:
No, CLI tools never pass a configuration object when creating a Forge.

Reply:
Good.